### DISCUSSION TITLE
Livewire legacy model binding throws LogicNotFoundException even when the rule is present.

### DISCUSSION THREAD

**Discussion by irfanm96** [livewire/livewire#8867] [7. Bugs (archived)]
### Livewire version

v3.5.6

### Laravel version

v11.23.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

This is an example app I created to reproduce the error

The app has a Model called `Post`, and it has a JSON column called `tags`

```php

<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Post extends Model
{
    protected $guarded = [];

    protected $casts = [
        'tags' => 'array'
    ];
}

```

There is a livewire component called `AddPostTags.php` which tries to assign multiple tags to the post (Not ideal, but just for example purpose only)

I have enabled Legacy model binding, and the livewire component is pretty simple too.

```php

<?php

namespace App\Livewire;

use App\Models\Post;
use Livewire\Component;

class AddPostTags extends Component
{
    public ?Post $post = null;

    public function rules()
    {
        return [
            'post.tags' => ['required'],
        ];
    }

    public function mount()
    {
        $this->post = Post::firstOrCreate();
    }

    public function render()
    {
        return view('livewire.add-post-tags');
    }
}


```

I have a rule in place for the tags property for the post. I hope it is self-explanatory. 

And I have a simple front-end, a select component which can select multiple tags. 

Note: I'm using wire.model.live so we can see the error immediately.

```html

<div style="display: grid;place-items: center;height: 100vh">

    <div>
        <div>
            You are going to add tags to post #{{ $post->id }}
        </div>

        <br>

        <select wire:model.live="post.tags" multiple>

            <option value="tag-1">Tag 1</option>

            <option value="tag-2">Tag 2</option>

            <option value="tag-3">Tag 3</option>

        </select>


        <br>

        <div style="margin-top: 10px">
            Selected tags : {{ collect($post->tags ?? ['-'])->join(', ') }}
        </div>
    </div>

</div>



```

This is how the page renders

![image](https://github.com/user-attachments/assets/84bd3606-d268-4ac1-8d02-80cfbc0bf815)

It works fine, when i just select `Tag 1`, when i try to select `Tag 2` while `Tag 1` is selected i get this error

![image](https://github.com/user-attachments/assets/331510d7-ae03-469b-be20-5ca333dff799)

The logic not found exception is thrown from the `hasRuleFor` method in 

.../src/Features/SupportLegacyModels/SupportLegacyModels.php:46

```php

    static function hasRuleFor($component, $path) {
        $path = str($path)->explode('.');

        $has = false;
        $end = false;

        $segmentRules = static::getRules($component);

        foreach ($path as $key => $segment) {
            if ($end) {
                throw new \LogicException('Something went wrong');
            }

            if (!is_numeric($segment) && array_key_exists($segment, $segmentRules)) {
                $segmentRules = $segmentRules[$segment];
                $has = true;
                continue;
            }

            if (is_numeric($segment) && array_key_exists('*', $segmentRules)) {
                $segmentRules = $segmentRules['*'];
                $has = true;
                continue;
            }

            if (is_numeric($segment) && in_array('*', $segmentRules, true)) {
                $has = true;
                $end = true;
                continue;
            }

            if (in_array($segment, $segmentRules, true)) {
                $has = true;
                $end = true;
                continue;
            }

            $has = false;
        }

        return $has;
    }


```

The same component works in V2, without any issues, i can confirm that.

There is a bug in the `hasRuleFor` function where it throws the error. `hasRuleFor` function returns the `$has` variable, and it is initialised to `false`, and if it was updated to `true` in those if conditions i think we should use `break` instead of `continue`. Logically since we do not set the `$has` variable to false, if it was set to true once in the loop, we can safely break the foreach and return.

And this fixes the issue too.

To reproduce the error, please try the following

1. Clone the repo https://github.com/irfanm96/livewire-legacy-model-binding
2. php artisan migrate (Default Sqlite will work, MySql not needed)
3. herd link / valet link
4. Visit the repo URL (http://livewire-legacy-model-binding.test/add-post-tags)
5. Select more than 1 tags
6. The exception will be thrown

I might be wrong thinking this is a bug, i did not create a PR for this, I would like to here from others if i have missed anything or making a mistake.

### Repository or Wirebox URL

https://github.com/irfanm96/livewire-legacy-model-binding

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by adamstamper** [comment]
Hi @irfanm96 - I've experienced this too, with nullable arrays. It's annoying. I'm also not entirely clear on the underlying logic there at the moment, but I _think_ your proposal is correct. For now I'm working around it by specifying validation rules that apply to each array item's content via `.*` - ie, in the `rules()` function having `'post.tags.*' => ['string']`. This only works for me because the array can be empty, unlike in your demo where at least one tag is required.

**Reply by mazedlx** [reply]
Thank you, this worked for me too.
